oxenc/bt_serialize.h

```diff
@@ -738,32 +738,34 @@ namespace detail {
 /// memory stays valid for the lifetime of the bt_list_consumer object.
 class bt_list_consumer {
   protected:
-    std::string_view data;
+    std::string_view data;            // Remaining data; this gets prefix-removed as we go
+    const char* start = data.data();  // Pointer to the start of the initial data, so that we can
+                                      // get the entire data when needed (e.g. for signatures)
     bt_list_consumer() = default;
 
+    struct load_tag {};
+    bt_list_consumer(std::string_view data, load_tag) : data{data} {}
+
   public:
-    bt_list_consumer(std::string_view data_) : data{std::move(data_)} {
+    bt_list_consumer(std::string_view data_) : bt_list_consumer{data_, load_tag{}} {
         if (data.empty())
             throw std::runtime_error{"Cannot create a bt_list_consumer with an empty string_view"};
         if (data[0] != 'l')
             throw std::runtime_error{"Cannot create a bt_list_consumer with non-list data"};
         data.remove_prefix(1);
     }
-    bt_list_consumer(std::basic_string_view<unsigned char> data_) :
+    bt_list_consumer(std::basic_string_view<unsigned char> data) :
             bt_list_consumer{
-                    std::string_view{reinterpret_cast<const char*>(data_.data()), data_.size()}} {}
-    bt_list_consumer(std::basic_string_view<std::byte> data_) :
+                    std::string_view{reinterpret_cast<const char*>(data.data()), data.size()}} {}
```

Comment:
I presume this change is safe with respect to shadowing?

Reply:
Yup.